app/src/androidTest/java/sdp/moneyrun/database/DatabaseProxyTest.java

```diff
@@ -39,38 +38,34 @@ public static void setPersistence(){
     }
     @Test
     public void getPlayerFromDatabase() {
-     //   FirebaseDatabase.getInstance().goOffline();
+       FirebaseDatabase.getInstance().goOffline();
         final Player player = new Player(1236, "Johann", 0);
         final PlayerDatabaseProxy db = new PlayerDatabaseProxy();
         CountDownLatch updated = new CountDownLatch(1);
         //adding to db
         CountDownLatch added = new CountDownLatch(1);
         OnCompleteListener listener = task -> added.countDown();
         db.putPlayer(player, listener);
-        try {
-            added.await(ASYNC_CALL_TIMEOUT, TimeUnit.SECONDS);
-            assertThat(added.getCount(), is(0L));
-        }catch (InterruptedException e){
-            fail();
-        }
-
         Task<DataSnapshot> testTask = db.getPlayerTask(player.getPlayerId());
         testTask.addOnCompleteListener(task -> {
             if(task.isSuccessful()) {
                assert( player.equals(db.getPlayerFromTask(testTask)));
                updated.countDown();
             }else{
+                System.out.println(task.getException());
                 assert (false);
             }
         });
-
+        while(!testTask.isComplete()){
```

Comment:
We can't use a countdown latch instead of busy waiting?

Reply:
Yes we can, it may be a survivor of the cleansing but the most probable is that with several people having worked on those classes this week, the merge was not absolutely perfect

Reply:
Alright sure!